clr.py

```diff
@@ -127,24 +133,42 @@ def compute_momentum(self):
             new_momentum = self.max_momentum
 
         elif self.clr_iterations > self.mid_cycle_id:
-            current_percentage = 1. - ((self.clr_iterations - self.mid_cycle_id) / float(self.mid_cycle_id))
-            new_momentum = self.max_momentum - current_percentage * (self.max_momentum - self.min_momentum)
+            current_percentage = 1. - (
+                (self.clr_iterations - self.mid_cycle_id) / float(
+                    self.mid_cycle_id))
+            new_momentum = self.max_momentum - current_percentage * (
+                self.max_momentum - self.min_momentum)
 
         else:
             current_percentage = self.clr_iterations / float(self.mid_cycle_id)
-            new_momentum = self.max_momentum - current_percentage * (self.max_momentum - self.min_momentum)
+            new_momentum = self.max_momentum - current_percentage * (
+                self.max_momentum - self.min_momentum)
 
         return new_momentum
 
     def on_train_begin(self, logs={}):
         logs = logs or {}
 
+        self.epochs = self.params['epochs']
+        self.batch_size = self.params['batch_size']
+        self.samples = self.params['samples']
+        self.steps = self.params['steps']
+
+        if self.steps is not None:
+            self.num_iterations = self.epochs * self.steps
+        else:
+            self.num_iterations = self.epochs * self.samples // self.batch_size
```

Comment:
Assumes samples is divisible by batch size. Should check for the remainder and add it if necessary.